packages/rancher-vsphere-cpi/charts/templates/vsphere-cpi-rbac.yaml

```diff
@@ -1,14 +1,15 @@
+{{- if .Values.cloudControllerManager.rbac.enabled -}}
 # Source: https://github.com/kubernetes/cloud-provider-vsphere/blob/master/releases/v1.19/vsphere-cloud-controller-manager.yaml
 apiVersion: v1
 kind: ServiceAccount
 metadata:
-  name: cloud-controller-manager
+  name: vsphere-cloud-controller-manager
```

Comment:
These all get updated on upgrade right, curious if any of these are immutable.

Reply:
Yeah this can be updated. Helm 3's logic is
- Run `helm template`
- For each resource in `helm template`, check if it exists. If not, create. If so, patch.
- If a resource that is currently annotated with this Helm release exists and the `keep` annotation is not specified and it was not updated or created in the prior step (e.g. does not exist in the template), delete it.

This logic is encoded in [the kube package Client.Update](https://github.com/helm/helm/blob/3f283c7c35fffda567894b9bf6cede5e19d43ffa/pkg/kube/client.go#L192-L198), which is [used by the Configuration object](https://github.com/helm/helm/blob/a499b4b179307c267bdf3ec49b880e3dbd2a5591/pkg/action/action.go#L366) and [is called on an Upgrade](https://github.com/helm/helm/blob/a499b4b179307c267bdf3ec49b880e3dbd2a5591/pkg/action/upgrade.go#L317-L321).

So in this case, this is identical to deleting the ServiceAccount in favor of a new one.